### PR TITLE
Clarify GenericColumnReader::read_records

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Closes #.

# Rationale for this change
 
https://github.com/apache/arrow-rs/pull/4376#issuecomment-1640245592

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: parquet/src/column/reader.rs [hunk 1]
```diff
@@ -212,14 +212,16 @@ where
         Ok((values, levels))
     }
 
-    /// Read up to `num_records` returning the number of complete records, non-null
-    /// values and levels decoded
+    /// Read up to `num_records` whole records, returning the number of complete
+    /// records, non-null values and levels decoded. Records will not be partially read
```
Comment (alamb) [inline parquet/src/column/reader.rs:216]: What does "partially read" mean? Like does it mean that the a record will not be read unless there is sufficient data for all columns?

If so, maybe

```suggestion
    /// records, non-null values and levels decoded. Records will not be partially read, 
    /// they are only read as long as there is sufficient data for all columns available. 
```

**Review by alamb — APPROVED** [review]
Thank you @tustvold  -- it is a great idea to clarify the documentation for future readers. 

I think clarifying what "partially read" means would help here (I left a suggestion but am not sure it is correct). 

cc @zeevm

File: parquet/src/column/reader.rs (hunk 1 above)
Comment (tustvold) [reply parquet/src/column/reader.rs:216]: You only have a single column here, this is referring to record shredding of repeated values, will clarify further...

**Review by alamb — APPROVED** [review]
Thank you

**Pull request merged**